doc/ntp/ntp-design.md

```diff
@@ -0,0 +1,898 @@
+        }
+    },
+    "NTP_SERVER": {
+        "1.1.1.1": {
+            "state": "disabled",
+            "version": "4",
+            "key": "42",
+            "iburst": "on",
+            "association_type": "server"
+        },
+        "somehost": {
+            "state": "enabled",
+            "version": "3",
+            "iburst": "on",
+            "association_type": "pool"
+        }
+    },
+    "NTP_KEY": {
+        "1": {
+            "type": "md5",
+            "value": "mypassword",
+            "trusted": "yes"
+        },
+        "42": {
+            "type": "sha1",
+            "value": "theanswer",
+            "trusted": "no"
+        }
+    }
+}
+```
+
+<!-- omit in toc -->
+### 2.4.4 Configuration migration
+No special handling is required
+
+## 2.5 Flows
+
+<!-- omit in toc -->
+### 2.5.1 NTP config flow
+![NTP config flow](images/ntp_config_flow.png "Figure 2: NTP config flow")
+<!-- omit in toc -->
+###### Figure 2: NTP config flow
+
+<!-- omit in toc -->
+### 2.5.2 NTP show status sequence
+![NTP show status sequence](images/ntp_show_status_flow.png "Figure 3: NTP show status sequence")
+<!-- omit in toc -->
+###### Figure 3: NTP show status sequence
+
+<!-- omit in toc -->
+### 2.5.2 NTP show configuration
+![NTP show configuration](images/ntp_show_configuration_flow.png "Figure 4: NTP show configuration")
+<!-- omit in toc -->
+###### Figure 4: NTP show configuration
+
+## 2.6 CLI
+
+<!-- omit in toc -->
+### 2.6.1 Command structure
+
+**User interface**:
+```
+config
+\-- ntp
+    |-- state <state>
+    |-- authentication <auth>
+    |-- dhcp <state>
+    |-- vrf <vrf>
+    |-- server
+    |   |-- add <host> SRV_OPTS
+    |   |-- del <host>
+    |    \- <host> state <state>
+     \- key
+        |-- add <key_id> <key_value> KEY_OPTS
+         \- del <key_id>
+
+show
+\-- ntp
+    |-- global
+    |-- server
+     \- key
+```
+
+**Options:**
+
+General:
+1. `<state>`, `<auth>` - state string: `enum: enabled, disabled`
+3. `<vrf>` - OOB VRF device: `enum: default, mgmt`
+3. `<host>` - NTP server: `host: ipv4, ipv6, hostname`
+4. `<key_id>` - Authentication key ID: `integer: 1-65535`
+
+_SRV_OPTS_:
+1. `-t|--type` - NTP server association type: `enum: server, pool`
+2. `-k|--key` - NTP authentication key ID: `integer: 1-65535`
+3. `-v|--version` - NTP protocol version `integer: 3-4`
+4. `-a|--aggressive` - NTP enable aggresive polling
+
+_KEY_OPTS_:
+1. `-t|--type` - NTP key type: `enum: md5, sha1`
+3. `-s|--trusted` - NTP key trustiness: `enum: yes, no`
+
+<!-- omit in toc -->
+### 2.6.2 Usage examples
+
+#### 2.6.2.1 Config command group
+
+**The following command enabled/disables NTP feature:**
+```bash
+config ntp state enabled
+config ntp state disabled
+```
+
+**The following command enabled/disables NTP authentication:**
+```bash
+config ntp authentication enabled
+config ntp authentication disabled
+```
+
+**The following command enabled/disables NTP DHCP client:**
+```bash
+config ntp dhcp enabled
```

Comment:
Could you please specify when NTP servers are acquired from dhcp as well as statically configured, what will be the final version of ntp.conf?

Reply:
DHCP will be the final.
If you enable DHCP - it will be used instead of statically configured